### PR TITLE
automatic install required pkgs for linux

### PR DESCRIPTION
**Pull request by mateeullahmalik**

### CODE REVIEW

File: cmd/install.go [hunk 1]
```diff
@@ -524,8 +524,26 @@ func checkInstalledPackages(ctx context.Context, tool constants.ToolType) (err e
...
+			Error("automatic install for required pkgs only set up for linux")
+
+		return fmt.Errorf("missing required pkgs: %s", strings.Join(notInstall, ","))
+	}
+
+	// install missing packages [only for linux at this time]
+	// TODO: devise a mechanism for installing pkgs for mac & windows
+	for _, pkg := range notInstall {
+		args := []string{"install", "-y", pkg}
+		out, err := RunCMD("apt", args...)
```
Comment (nguyen202) [inline cmd/install.go:539]: I think need `sudo` permission?

**Review by a-ok123 — APPROVED** [review]

File: cmd/install.go [hunk 2]
```diff
@@ -524,8 +524,35 @@ func checkInstalledPackages(ctx context.Context, tool constants.ToolType) (err e
...
+			Error("automatic install for required pkgs only set up for linux")
+
+		return fmt.Errorf("missing required pkgs: %s", pkgsStr)
+	}
+
+	return installMissingReqPackagesLinux(notInstall)
+}
+
+func installMissingReqPackagesLinux(pkgs []string) error {
+	log.WithField("packages", strings.Join(pkgs, ",")).
```
Comment (nguyen202) [inline cmd/install.go:545]: nit: missing WithContext(ctx) for log lines.

File: cmd/install.go [hunk 3]
```diff
@@ -524,8 +524,35 @@ func checkInstalledPackages(ctx context.Context, tool constants.ToolType) (err e
...
+		Info("system will now install missing packages")
+
+	for _, pkg := range pkgs {
+		out, err := RunCMD("sudo", "apt", "install", "-y", pkg)
```
Comment (nguyen202) [inline cmd/install.go:549]: can log pkg we're trying to install?

Comment (mateeullahmalik) [reply cmd/install.go:549]: already logged above. 
if any pkg fails, it will display a proper log message

Comment (a-ok123) [reply cmd/install.go:549]: you can add pkg names to 		Info("system will now install missing packages")

**Review by a-ok123 — APPROVED** [review]

**Review by nguyen202 — APPROVED** [review]
LGTM!

**Pull request merged**